src/products/api/ProductDto.ts

```diff
@@ -0,0 +1,25 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { Product } from '../../model/product.entity';
+
+export class ProductDto {
+  @ApiProperty()
+  name: string;
+
+  @ApiProperty()
+  category: string;
+
+  @ApiProperty()
+  photoUrl: string;
+
+  @ApiProperty()
+  description: string;
+
+  public static covertToApi(c: Product): ProductDto {
+    return {
+      name: c.name,
+      category: c.category,
+      photoUrl: c.photoUrl,
+      description: c.description,
+    };
+  }
```

Comment:
Extract to either the controller or a separate class. Avoid using such static factory methods, it's bad practice.  Also, you have to return an instance of `ProductDto`